src/hotspot/cpu/s390/gc/shared/barrierSetAssembler_s390.cpp

```diff
@@ -171,7 +171,7 @@ void BarrierSetAssembler::try_resolve_jobject_in_native(MacroAssembler* masm, Re
 
 void BarrierSetAssembler::nmethod_entry_barrier(MacroAssembler* masm) {
   BarrierSetNMethod* bs_nm = BarrierSet::barrier_set()->barrier_set_nmethod();
-  __ align(4, __ offset() + PATCHABLE_BARRIER_VALUE_OFFSET); // must align the following block which requires atomic updates
+  __ align(4, __ offset() + BARRIER_TOTAL_LENGTH); // must align the following block which requires atomic updates
```

Comment:
Now, this is confusing. We don't want to align the end of the barrier. We need to align the patchable field which is the immediate field of the cfi instruction.